### PR TITLE
Added dhcp-snooping feature.

### PR DESCRIPTION
**Pull request by burnyd**
## Change Summary

Update from issue #1090 for dhcp snooping.

## Types of changes
- [X] New feature (non-breaking change which adds functionality)

## Component(s) name

`arista.avd.eos.cligen`

## Proposed changes
Add dhcp snooping feature.

## How to test
Tested within molecule included files.

## Checklist:
- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-dhcp-snooping.j2 [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% if ip_dhcp_snooping.enabled is arista.avd.defined(true) %}
+!
+ip dhcp snooping
+| VLAN |
+| ---  |
+{%   for vlan in ip_dhcp_snooping.vlans | arista.avd.natural_sort %}
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-dhcp-snooping.j2:6]: Maybe we could add a second column to this table with the state of DHCP snooping. Current documentation provide list of cli in addition to the CLI include.

**Review by titom73 — CHANGES_REQUESTED** [review]
Thanks for your PR! Please see comment inline related to documentation rendering.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-dhcp-snooping.j2 (hunk 1 above)
Comment (burnyd) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-dhcp-snooping.j2:6]: @titom73 can you provide an example either in one of the other documentation jinja templates or likewise?

Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-dhcp-snooping.j2:6]: @burnyd you can look at `vlans.j2` in the documentation folder. In general, we use a table with at least 2 columns with this approach. The CLI part is done with the `include` in the second part of your file:


```jinja2
## VLANs Summary

| VLAN ID | Name |
| ------- | ---- |
{%     for vlan in vlans | arista.avd.natural_sort %}
{%         set row_name = vlans[vlan].name | arista.avd.default('-') %}
| {{ vlan }} | {{ row_name }} |
{%     endfor %}
```

**Comment by carlbuchmann** [comment]
@burnyd - closing PR as it's stale, feel free to re-submit a new PR if you require this feature!

**Pull request closed without merging**